enhancements/kube-apiserver/stability-api-removal-notifications.md

```diff
@@ -0,0 +1,162 @@
+---
+title: api-removal-notifications
+authors:
+  - "@sanchezl"
+reviewers:
+  - "@deads2k"
+  - TBD
+approvers:
+  - TBD
+creation-date: 2021-02-01
+last-updated: 2021-02-01
+status: provisional
+---
+
+# API Removal Notifications
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Notify customers that an API that will be removed in the next release is in use.
+
+## Motivation
+
+Kubernetes is being more aggressive with the removal of APIs that have been deprecated or pre-release APIs that have
+reached a dead-end and will never reach GA. Additionally, the ability to re-enable a previously removed API has been
+removed.
+
+Customers should be notified that an API that will be removed in the next release is in use so that they can decide if
+there is a need to take action in order to ensure success when upgrading to the next release.
+
+### Goals
+
+1. Notify customers that an API that will be removed in the next release is in use.
+2. Notify EUS customers that an API that will be removed in the next EUS release is in use.
+3. Document how to identify the workload that is using the API to be removed.
```

Comment:
I want a section exploring at least two possibilities of how this could be done at an API level so we don't have to have a manual process.  I think I'll lose my mind trying to fix these if I have to manually dig into audit logs.